languages/rust/oso/tests/test_oso.rs

```diff
@@ -0,0 +1,161 @@
+use oso::PolarClass;
+use std::path::{Path, PathBuf};
+
+mod common;
+
+use common::OsoTest;
+
+fn test_file_path() -> PathBuf {
+    let path = Path::new(env!("CARGO_MANIFEST_DIR"));
+    path.join(Path::new("tests/test_oso.polar"))
+}
+
+#[derive(PolarClass, Debug, Clone, PartialEq)]
+struct Actor {
+    #[polar(attribute)]
+    name: String,
+}
+
+impl Actor {
+    pub fn new(name: String) -> Self {
+        Self { name }
+    }
+
+    pub fn companies(&self) -> Vec<Company> {
+        vec![Company { id: 1 }]
+    }
+}
+
+#[derive(PolarClass, Debug, Clone, PartialEq)]
+struct Widget {
+    #[polar(attribute)]
+    id: i64,
+}
+
+impl Widget {
+    pub fn new(id: i64) -> Self {
+        Self { id }
+    }
+}
+
+#[derive(PolarClass, Debug, Clone, PartialEq)]
+struct Company {
+    #[polar(attribute)]
+    id: i64,
+}
+
+impl Company {
+    pub fn new(id: i64) -> Self {
+        Self { id }
+    }
+
+    pub fn role(&self, actor: Actor) -> String {
+        if actor.name == "president" {
+            "admin".to_string()
+        } else {
+            "guest".to_string()
+        }
+    }
+}
+
+fn test_oso() -> OsoTest {
+    let mut test = OsoTest::new();
```

Comment:
I would probably skip the use of `OsoTest` for this file, since this file is more about testing the publicly-facing part of Oso. The main purpose of `OsoTest` is to expose a bunch of useful helpers for unit testing functionality.